ui/Model.elm

```diff
@@ -126,42 +154,113 @@ getReleaseList =
         |> Kinto.send ReleasesFetched
 
 
-extractChannels : List Release -> Dict.Dict Channel Bool
-extractChannels releaseList =
-    List.map (.details >> .channel >> (\c -> ( c, True ))) releaseList
+extractFilterSet : (Release -> String) -> List Release -> Dict.Dict String Bool
+extractFilterSet accessor releases =
+    List.map (accessor >> (\c -> ( c, True ))) releases
         |> Dict.fromList
 
 
-filterReleases : Model -> List Release
-filterReleases { filters, releases } =
+extractVersions : List Release -> Dict.Dict String Bool
+extractVersions releases =
+    -- TODO: sort desc
+    let
+        extractVersion ( version, x ) =
+            case (String.split "." version) of
+                v :: _ ->
+                    ( v, x )
+
+                _ ->
+                    ( "None", x )
+    in
+        extractFilterSet (.details >> .filename) releases
+            |> Dict.toList
+            |> List.map extractVersion
+            |> Dict.fromList
```

Comment:
You mean contributing a `Dict.mapKeys` to Elm core? That's not something I'm planning to do anytime soon, mostly for laziness purpose. But feel free to work on this!

If you mean creating a local `mapDictKeys` function, I don't see the benefit as it's the only place where we'd need it.